WooCommerce/Classes/ViewRelated/Search/Product/ProductSearchUICommand.swift

```diff
@@ -101,6 +104,11 @@ final class ProductSearchUICommand: SearchUICommand {
                 onCompletion?(true)
                 return
             }
+            // Skips the product search API request if the keyword is empty.
+            guard keyword.isNotEmpty else {
+                onCompletion?(true)
+                return
+            }
```

Comment:
I'm not sure if this is what we want to do. Consider this case: when there's a filter applied to the list in the Products tab, entering and then clearing the keyword in the search screen will return only the filtered items and the results. Do we want to show the whole list instead?

https://user-images.githubusercontent.com/5533851/192682711-bee4f16e-9f38-4342-87ff-31143c8eb598.mp4

If this is the intended behavior, please ignore this comment.

Reply:
Before this change: with an empty search query, launching the initial product search screen does not trigger any API requests. However, if you tap on the SKU tab, it still triggers an API request to fetch products and that's why a bunch of products appear suddenly. I felt this UX is kind of confusing where other products start appearing after switching to the SKU tab, even though it does load other products.

With this change, at least the product list remains the same in both All Products and SKU tabs when the search query is empty. It's a tradeoff where products outside of the filters aren't shown, but if we want to show all products I think we should implement it intentionally if we're fine with the performance impact. Since this impacts merchants with filters applied on the products tab and probably not all merchants do so, I thought it's better to optimize for the consistent product list in both search tabs. It'd be great to get data on how many merchants actually use filters on the products tab, I'll take a look soon.

I hope this explains the change more in addition to the PR description, please let me know if you have any thoughts on this behavior!

Reply:
Sounds good, thanks for the explanation!